hub/core/chunk.py

```diff
@@ -47,6 +50,49 @@ def __init__(
 
         self._data: Union[memoryview, bytearray] = data or bytearray()
 
+        # Decompressed caches
+        self._decompressed_samples: Optional[List[np.ndarray]] = None
+        self._decompressed_data: Optional[bytes] = None
+
+    def decompressed_samples(
+        self,
+        compression: Optional[str] = None,
+        dtype: Optional[Union[np.dtype, str]] = None,
+    ) -> List[np.ndarray]:
+        """Applicable only for compressed chunks"""
+        if self._decompressed_samples is None:
+            shapes = [
+                self.shapes_encoder[i] for i in range(self.shapes_encoder.num_samples)
+            ]
+            if compression == "lz4":
```

Comment:
why do we have this hardcoded just for lz4? zstd and other compressions will also probably function similarly. we should probably have a constant list and check in it.